src/main/java/frc/robot/subsystems/auto/AutoAlgaeElevatorCommand.java

```diff
@@ -0,0 +1,97 @@
+package frc.robot.subsystems.auto;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.ElevatorSubsystem;
+import frc.robot.subsystems.drivebase.CommandSwerveDrivetrain;
+import java.util.List;
+
+/**
+ * Command that automatically moves the elevator to the proper height for the nearest algae based on
+ * alliance and current robot pose.
+ */
+public class AutoAlgaeElevatorCommand extends Command {
+
+  private final CommandSwerveDrivetrain drivebase;
+  private final ElevatorSubsystem elevator;
+
+  private static final AprilTagFieldLayout aprilTagFieldLayout =
+      AprilTagFieldLayout.loadField(AprilTagFields.k2025ReefscapeWelded);
+
+  private static final List<Pose2d> blueAlgaePoses =
+      List.of(
+          aprilTagFieldLayout.getTagPose(17).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(19).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(21).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(18).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(20).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(22).get().toPose2d());
+
+  private static final List<Pose2d> redAlgaePoses =
+      List.of(
+          aprilTagFieldLayout.getTagPose(6).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(8).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(10).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(7).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(9).get().toPose2d(),
+          aprilTagFieldLayout.getTagPose(11).get().toPose2d());
+
+  public AutoAlgaeElevatorCommand(CommandSwerveDrivetrain drivebase, ElevatorSubsystem elevator) {
+    this.drivebase = drivebase;
+    this.elevator = elevator;
+    addRequirements(elevator);
+  }
+
+  private static boolean isBlue() {
+    return DriverStation.getAlliance()
+        .map(alliance -> alliance.equals(Alliance.Blue))
+        .orElse(false);
+  }
+
+  private static Pose2d getNearestAlgae(Pose2d robotPose) {
+    List<Pose2d> poses = isBlue() ? blueAlgaePoses : redAlgaePoses;
```

Comment:
We do this a bunch in different functions.  We should add this to the class instantiation and do it once instead of for every time a function is called.